### ISSUE TITLE
Commercial use

### ISSUE THREAD

**Issue by kasual1** [mattdesl/color-wander#22]
Hey mattdesl,

first of all thanks for so much inspiration! Your work is truely amazing.

I'm interested in using the output of color-wander for commercial use. What am I allowed to do? And what am I not allowed to do?

E.g. would it be fine if I modify the rendering and drawing algorithm to fit my own needs and also use my own images for the distortion maps?

Thanks in advance!

**Comment by mattdesl** [comment]
Hi there, the output is not licensed for commercial use, if you'd like to use it for something commercial perhaps the best way would be to chat about a licensing fee for the software, and for use within your specific application. If you're interested in purchasing that we could chat further over [email](info@mattdesl.studio).

**Comment by 1j01** [comment]
@mattdesl What intellectual property do you think you have over the output of the program?

**Comment by mattdesl** [comment]
IANAL but I have not seen any clear law defining the particular topic of generative art, so the following is my own interpretation.

In simple terms: I created a software to produce images of a specific style and design, and thus the IP also extends to the output of the software. 

Imagine this analogy: you happen upon the entire source code of the game No Man’s Sky. You compile the code and it produces a totally random build of the game. This does not mean you now magically own the IP of the game (ie: the output of the code), nor do you own the IP of its procedurally generated designs, nor even do you own the IP of screenshots you take of your build of the game. 

The only way you could use the screenshots/outputs is in fair use, e.g. reviewing or sharing No Man’s Sky on social media.

Ultimately the code here is MIT so that people can copy bits and pieces and introduce them into OSS in ways that may or may not be related to generative art. However the output and artistic work here is not intended to be MIT as I would rather not have it easily forked and resold as prints or published under a different artist’s name.

PS: It goes without saying that significant modifications to the source and design would be considered a derivative work and thus shouldn’t be a problem.

**Comment by ff6347** [comment]
Interesting topic. My 50 cent? (I'll keep on writing even if you don't want to hear it 😉 )
@mattdesl IMO Your analogy to the source code of No Man's Sky would not hold up in court due to the fact that it is not published under an OSS license. I also think that if you want permit commercial use you should not use a MIT License. Currently I would say people can do what they want with your code. If you want more control over the results you should choose another license. Maybe [CC](https://creativecommons.org/)?
I really would like to hear from a lawyer about this topic.

Edit: 

Sorry didn't see your remark in the README

>  the actual design/visuals/concept/artwork is licensed under Creative Commons Attribution-NonCommercial-NoDerivatives 4.0 International.

Still I'm not sure how this would hold up in court. How to define where the "pure code" ends and the "design/visuals/concept/artwork" starts?

**Comment by mattdesl** [comment]
I think it would be hard to argue one way or the other as there doesn’t seem to be a precedent set around generative art, and my wording of the license is muddy. 

If somebody downloads screenshots of the website and tries to sell it as an art print, I feel it would be pretty clearly against the CC license and probably not very hard to argue in court that the screenshot is within the design/visuals/artwork.

Conversely, the wording right now probably does not inspire confidence in companies looking to copy or reuse bits of the code—I imagine a company like Google would avoid using the code here for anything, as the license is too muddy. “Design/concept” is a broad term and could maybe, if argued, even encompass the entire project/software.

Since IANAL this is all conjecture. If a lawyer steps and and suggests it’s too fragile, then I’ll just remove the MIT part of the license and the code will no longer be reusable in any form. 🤷‍♂️ Or, I could explore a non-commercial Polyform license. 

Maybe @kemitchell would have more insight on this.

**Comment by boulabiar** [comment]
My 0.02¢.
Legally, if something is licensed with an MIT licence, you can do anything with it. That's the interpretation of the main 2 authorities of free software (FSF) and open source (OSI ). No descrimination can be made on the use of the software or what is generated from it.
Mongodb tried to create a license forbidding the use of the database in the cloud by AWS and got rejected by both FSF and OSI. The packages got even pulled from some distributions.
Simply you can't have free software and in the same time forbidding users from doing what they want with the code, it breaks the 4 liberties.
Even Autodesk software or Adobe have generative plugins and yet don't (or can't) force users to license the output of the program.

For some free software like Qt, they have GPL/Commercial licensing. 

For music it's then creative commons licenses which they include a non mixing case.

In the case of generative tools which make users pay for the output, the model used is to host the software on a server and not open sourcing it. Create a low resolution version of the output, then the user pay for the high resolution. Then the user can do whatever he wants with the video/image.

For the high resolution version downloaded from a server, you can add a condition for CC BY no remix, but that's not open source.

Meeting free software people can be good in conferences like Fosdem in Brussels, because I feel like artists are not in phase with the philosophy of open source and that needs a discussion from both parts.

**Comment by mattdesl** [comment]
@boulabiar I agree that if this project were to be simply MIT licensed, I would hold no claim over the output. However, this is not MIT licensed: it is a "fudged" and made-up license that I have created, which is in part MIT (explicitly the software), and in part CC (explicitly the visuals/design/output). So, it is no longer MIT license, and shouldn't be considered as an MIT licensed project. That's how I see this, at least.

**Comment by Luxapodular** [comment]
Hey @mattdesl and all, this is a great discussion! I'm curious as to why you licensed the code and output under separate licenses, rather than apply say the CC Attribution-NonCommercial license to the entire project? 

Is this specifically with the intention of allowing users to modify the code to produce a similar visual output and own those new creations? Or is it more along the lines of expecting someone may take bits and pieces of your code an integrate them into their own projects?

A license which allows someone to do this is definitely needed. I currently teach creative coding classes, and applying a license which allows manipulation of source code with an expectation of a significant "transformation" of the output is usually something I apply through assignment guidelines.

**Comment by mattdesl** [comment]
@Luxapodular The reasons are twofold – (1) CC should generally not be applied to code/software (even Creative Commons recommends against it), and (2) like you say, I want to give others the ability to extract some of the code in this project and re-publish freely it under a different context, such as an npm module for random colour picking. Along those lines, if somebody forks and significantly remixes this project in such a way that they create an original and derivative artwork, I'd also be happy to see that.

**Comment by Luxapodular** [comment]
Makes sense. I guess the most difficult part here is defining how much remixing / visual differentiation actually constitutes an original and derivative work. 

As it stands, it does seem that the mixed license you've applied is a bit contradictory, as applying separate licenses to the code and output of that code (Other than a specific instance of that output) doesn't seem like it would actually work. But then again, I am not a lawyer!

I'll come back to this discussion after having a chat with a few other professors/digital artists and see if they've found a solution to this.

**Comment by edowado** [comment]
The whole idea of licensing digital art (or art in general) is ridiculous to me, what happens if I make something that looks likes something someone else made? 
I know that I'm not contributing much to this conversation but There 👏 is 👏 not 👏 such 👏 thing 👏 as 👏 intellectual 👏 property.
Having said that and before you start hating me I would never use something someone else made for profit but I would never restrict someone from doing it also.
[Here is a cool quote from Jim Jarmusch](https://www.goodreads.com/quotes/131591-nothing-is-original-steal-from-anywhere-that-resonates-with-inspirationl)
